com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -886,12 +886,22 @@ void ResetData()
         /// <seealso cref="IActionReceiver.OnActionReceived"/>
         public virtual void Heuristic(in ActionBuffers actionsOut)
         {
+            var brainParams = m_PolicyFactory.BrainParameters;
+            var actionSpec = brainParams.ActionSpec;
+            // For continuous and discrete actions together, we don't need to fall back to the legacy method
+            if (actionSpec.NumContinuousActions > 0 && actionSpec.NumDiscreteActions > 0)
```

Comment:
Better to avoid any confusion here - don't call the float[] version if we're using continuous + discrete (plus avoid munging the arrays)